commit 92ba3aec4e882d7dc1793224d1c5cb65f165688d (refs/changes/36/4236/2)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2018-06-22T19:32:56+02:00 (1 year, 4 months ago)
    
    joyent/node-zkstream#33 `Client#isConnected()` throws TypeError
    Reviewed by: Alex Wilson <alex.wilson@joyent.com>

diff --git a/lib/client.js b/lib/client.js
index bbf0a9d..bb76ef7 100644
--- a/lib/client.js
+++ b/lib/client.js
@@ -3,7 +3,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  *
- * Copyright (c) 2016, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 module.exports = { Client: ZKClient };
@@ -213,7 +213,7 @@ ZKClient.prototype._newSession = function () {
 };
 
 ZKClient.prototype.isConnected = function () {
-	var conn = this.getConnection();
+	var conn = this.currentConnection();
 	return (conn !== undefined && conn.isInState('connected'));
 };
 
diff --git a/package.json b/package.json
index bc72a66..2b8146c 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "zkstream",
-  "version": "0.11.0",
+  "version": "0.11.1",
   "description": "",
   "repository": {
     "type": "git",
diff --git a/test/basic.test.js b/test/basic.test.js
index af4fad0..9b1a265 100644
--- a/test/basic.test.js
+++ b/test/basic.test.js
@@ -102,6 +102,7 @@ mod_tape.test('simple connect and ping', function (t) {
 	});
 
 	zkc.on('connect', function () {
+		t.ok(zkc.isConnected());
 		zkc.ping(function (err) {
 			t.error(err);
 			pinged = true;
@@ -121,6 +122,7 @@ mod_tape.test('double ping', function (t) {
 
 	zkc.on('close', function () {
 		t.strictEqual(pinged, 2);
+		t.notOk(zkc.isConnected());
 		t.end();
 	});
 
